pkg/utils/utils.go

```diff
@@ -59,6 +59,15 @@ var (
 	PublicStorageProviderID = "7993447f-687f-490d-875c-ac95e89a62a4"
 	// PublicStorageSpaceID is the space id used by the sharestorageprovider
 	PublicStorageSpaceID = "7993447f-687f-490d-875c-ac95e89a62a4"
+	// VirtualSpaceOwner is the non-existing user which is needed for public share impersonation
+	VirtualSpaceOwner = userpb.User{
+		Id: &userpb.UserId{
+			OpaqueId: "dbdf733d-427f-4ac3-8093-2cba5c307e4c",
+			Type:     userpb.UserType_USER_TYPE_INVALID,
+		},
+		DisplayName: "Virtual Space Owner",
+		Username:    "virtualowner",
```

Comment:
This Username value can be returned by `mustGetIdentifiers()` so callers will see it somewhere.

Whatever its string value is ("virtualowner", "nobody", "somethingelse"...), what will happen if there is a real user with the same value for Username?

Is it OK if the Username attribute is the same for multiple users? (And the `OpaqueId`, of course, is different - unique for every user)

Reply:
That is the same challenge for all other use cases with users, not only for this one 😄

Reply:
for the ocs api I think oc10 already blocks `public` as a username when accessing via public link: https://github.com/owncloud/core/blob/eebd754771a99bc44820eb67a6d57cea238e6b86/apps/dav/lib/Files/PublicFiles/PublicSharingAuth.php#L148

so we could use 'public', however that does not reflect the concept of a project space not having an owner. 

Not that I think of it ... won't this be a security risk if any public link will become the owner of all project spaces?

Reply:
ok, nevermind. the virtual user is not used to implement public link access, yet...

Reply:
internally all permission checks are done one user id, not on username. The ocs api thinks users are identified by username and expectst them to be uniqe. Simplest case: searching for users.

We can move to userids by either extending ocs or (my preferred choice) flesh out libregraph ...

Reply:
For reference, oC10 has "special" prohibited usernames:
https://github.com/owncloud/core/blob/master/lib/public/User/Constants.php

The oC10 username-namespace is polluted with all sorts of things.